src/shared/parameter_v2/examples/generated_dynamic_parameters.hpp

```diff
@@ -0,0 +1,351 @@
+    {
+        return std::const_pointer_cast<const NumericParameter<float>>(example_int_param);
+    }
+
+    const std::shared_ptr<NumericParameter<float>> mutableExampleFloatParam()
+    {
+        return example_float_param;
+    }
+
+    const std::shared_ptr<const Parameter<std::string>> exampleStringParam() const
+    {
+        return std::const_pointer_cast<const NumericParameter<std::string>>(example_string_param);
+    }
+
+    const std::shared_ptr<Parameter<std::string>> mutableExampleStringParam()
+    {
+        return example_string_param;
+    }
+
+    const std::shared_ptr<const EnumeratedParameter<RefereeCommand>> exampleEnumParam() const
+    {
+        return std::const_pointer_cast<const EnumeratedParameter<RefereeCommand>>(example_enum_param);
+    }
+
+    const std::shared_ptr<EnumeratedParameter<RefereeCommand>> mutableExampleEnumParam()
+    {
+        return example_enum_param;
+    }
+
+    const std::shared_ptr<const foo> fooConfig()
+    {
+        return std::const_pointer_cast<const foo>(foo_config);
+    }
+
+    const std::shared_ptr<foo> mutableFooConfig()
+    {
+        return foo_config;
+    }
+
+    const std::string name() const
+    {
+        return "example"
+    }
+
+    bool loadFromCommandLineArguments(int argc, char **argv) {
+        struct commandLineArgs {
+            bool help = false;
+            bool example_bool_param = true;
+            int example_int_param = 3;
+            uint example_uint_param = 3;
+            float example_float_param = 4.04;
+            std::string example_string_param = "Hello World";
+            RefereeCommand example_enum_param = RefereeCommand.HALT;
+        }
+
+        commandLineArgs args;
+        boost::program_options::options_description desc{"Options"};
+
+        desc.add_options()("help,h", boost::program_options::bool_switch(&args.help),
+                        "help screen");
+
+        desc.add_options()("example_bool_param", boost::program_options::value<bool>(&args.example_bool_param), "Can be true or false");
+        desc.add_options()("example_int_param", boost::program_options::value<int>(&args.example_int_param), "Can be any integer value in the range [min, max]");
+        desc.add_options()("example_uint_param", boost::program_options::value<uint>(&args.example_uint_param), "Can be any integer value in the range [min, max]. The minimum value must be >= 0");
+        desc.add_options()("example_float_param", boost::program_options::value<float>(&args.example_float_param), "Can be any value in the range [min, max] For C code, this will be generated as a float. For C++ code, this will be generated as a double.");
+        desc.add_options()("example_string_param", boost::program_options::value<std::string>(&args.example_string_param), "Any string value");
+        desc.add_options()("example_enum_param", boost::program_options::value<RefereeCommand>(&args.example_enum_param), "Accepts enum values from the specified enum.");
```

Comment:
Thoughts on generating enum values as `boost::program_options::value<std::string>` and then do `this->mutableExampleEnumParam()->setValue(fromStringToRefereeCommand(args.example_enum_param))`?

Reply:
Sounds good, done